python/python/lance/dataset.py

```diff
@@ -628,6 +628,12 @@ def alter_columns(self, *alterations: Iterable[Dict[str, Any]]):
                 nullability is not changed. Only non-nullable columns can be changed
                 to nullable. Currently, you cannot change a nullable column to
                 non-nullable.
+            - "data_type": pyarrow.DataType, optional
+                The new data type to cast the column to. If not specified, the column
+                data type is not changed.
+
+        Columns that are renamed can keep any indices that are on them. However, if
+        the column is casted to a different type, it's indices will be dropped.
```

Comment:
```suggestion
        the column is cast to a different type, its indices will be dropped.
```
`casted` might be ok too, I don't actually know for sure.